railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -68,9 +68,12 @@ def app
     end
 
     def bin
-      directory "bin" do |content|
+      directory "bin" , exclude_pattern: /\/bin\/yarn$/ do |content|
```

Comment:
There might be a cleaner way to do this, but after delving deep into the internals of app_generator; the code seemed to make heavy assumptions that files in the `bin` folder were to have `#!/usr/bin/env ruby` shebangs. Instead of refactoring, I opted to just take the simple, conditional path. If anyone knows a better solution, let me know :smile: